openpype/pipeline/farm/patterning.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+import re
+
+
+def match_aov_pattern(self, app, render_file_name):
+    """Matching against a AOV pattern in the render files
+    In order to match the AOV name
+    we must compare against the render filename string
+    that we are grabbing the render filename string
+    from the collection that we have grabbed from exp_files.
+    """
+    if app in self.aov_filter.keys():
+        for aov_pattern in self.aov_filter[app]:
+            if re.match(aov_pattern, render_file_name):
+                preview = True
+                return preview
```

Comment:
I don't really understand why we're currently putting this into its own file, especially because it still uses `self` as if it's on the class - it's not self-contained enoug with its arguments to belong on its own. Why are we making this so much more 'complicated'?

Reply:
Thanks Roy, the idea is that we're going about refactoring some stuff to reorganize the code a bit, it's being used in a couple of places in that file and it will be reused in some other submitters as well iirc the discussion that took place. 

`self` being passed might present an issue later on but it's not necessarily breaking oop principles, for now imho.

Reply:
If we're moving it to a separate file we should clean this up and make it generic enough to be usable - otherwise it's just a separate confusing part that we'll be left unused. I feel like we should only move it out if we're also refactoring other uses (e.g. not in the same plug-in).

Currently that'd maybe be `collect_render` for After Effects? (And that doesn't store it as `self.aov_filter` so the logic doesn't hold?

In essence this function does this:
```python
def match_patterns(patterns, value):
    if not patterns:
        return False
    return any(re.match(pattern, value) for pattern in patterns)
```

Could likely even be optimized to:
```python
def match_patterns(patterns, value):
    return re.match("|".join(patterns), value)
```

With the usage in the other file being:
```python
aov_filter_patterns = self.aov_filter.get(app, [])
match_patterns(aov_filter_patterns, render_file_name)
```

I don't see why we are currently in dire need to refactor this logic and warrant having it out in its own file, especially with how non-generic the code currently is.

Reply:
Ah, I see and understand your point.  For the logic, yes indeed it's doing the exact same thing, and even though that's more pythonic, it seems a little difficult to read for me

As for the refactoring, I think the guys have a better understanding of what's happening with the refactoring

Reply:
I'm totally for code re-use and refactoring to avoid code duplication. My point mostly is that the current state of this refactor doesn't make my sense and I'd hold of merging until cleaned.

Reply:
Thank you, I'm waiting for the guys to give it a looksie, I do lack the thorough understanding of the codebase to kind of segregate things out further, but from what I understood from the discussion the exact bit is to be reused in an identical way in a couple other places (checking if in context) and so on, so that's why I approached it like such, otherwise I really want to take out the contextual aspect of it

Reply:
> "I think it can be used even outside the deadline - in royal render module or in muster (if resurrected)"

If that's the case - then lets set it up like that. 

> And in that case signature of externalized function def match_aov_pattern(app, aov_patterns, render_file_name) might be more useful, as it would provide you more reusability.

Exactly this - and only if the function body has some complexity to it that requires consistency in multiple areas. If it's just a matter of regex matching I'm not sure separating now is the best time, but if we expect more complexity over time. Sure.

However if just a preparating I'd opt for adding it as a separate function within the Plugin file until it is actually used elsewhere. Unless we expect to directly create another PR to actually use that lib function. But that'd just be a personal preference just so we don't create clutter where we don't need it yet.

Reply:
Hey guys, thanks a lot for the notes.  We could potentially skip `aov_filter_patterns = self.aov_filter.get(app, [])` through directly using `aov_patterns = self.aov_filter.keys()` outside the function and `aov_pattern in aov_patterns[app]` within it as such:

```
for aov_pattern in aov_patterns[app]:
        if not aov_patterns:
            return False
        else:
            return any(re.match(aov_pattern, render_file_name) for aov_pattern in aov_patterns)   ```


What do you think?

And additionally testing within the for loop for it even existing through `if aov_pattern` would perhaps be a good idea too

Any input would be helpful!

Reply:
Ah, I see!  Thanks man I appreciate it!